transport/src/main/java/io/netty/channel/pool/ChannelHealthChecker.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.pool;
+
+import io.netty.channel.Channel;
+import io.netty.channel.EventLoop;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.Promise;
+
+/**
+ * Called before a {@link Channel} will be returned via {@link ChannelPool#acquire()} or
+ * {@link ChannelPool#acquire(Promise)}.
+ */
+public interface ChannelHealthChecker {
```

Comment:
Is there a way to do this periodically, but not every time you acquire a channel? I'm thinking about something like sending an application-level message to check if the other side is responsive (on top of verifying ``channel.isActive()``), which you wouldn't like to send so frequently.